common/membership/hostinfo.go

```diff
@@ -72,6 +81,39 @@ func (hi HostInfo) GetAddress() string {
 	return hi.addr
 }
 
+// GetNamedAddress returns the ip:port address
+func (hi HostInfo) GetNamedAddress(port string) (string, error) {
+	if port, set := hi.portMap[port]; set {
+		return fmt.Sprintf("%s:%d", hi.ip, port), nil
```

Comment:
I suggest using https://pkg.go.dev/net#JoinHostPort here. It is symmetrical to our usage `net.SplitHostPort` and also plays nicely with IPv6

Reply:
+1